cms/extensions/models.py

```diff
@@ -75,8 +81,19 @@ def copy_to_public(self, public_object, language):
         if public_extension:
             this.pk = public_extension.pk  # overwrite current public extension
             this.public_extension = None  # remove public extension or it will point to itself and raise duplicate entry
+
+            # Set public_extension concrete parents PKs. See issue #5494
+            for parent, field in this._meta.parents.items():
+                if field:
+                    setattr(this, parent._meta.pk.attname, getattr(public_extension, parent._meta.pk.attname))
```

Comment:
Actually @czpython your proposal is also more than 80 chars (95), and I'm not sure what is the benefit.

Instead i can change it like that:

    if field:
        setattr(this,
                parent._meta.pk.attname,
                getattr(public_extension, parent._meta.pk.attname))

But I'm not sure if you like this way of calling functions.

What you think?